packages/server-info/src/user/current.ts

```diff
@@ -97,13 +97,19 @@ export const patchCurrent = (
                 return;
             }
 
-            const updatedUser = await database.user.setNickname(user.id, trimmedNickname);
+            const updatedUser = await database.user.setProfileInfo(user.id, trimmedNickname, picture);
+            outputUser = convertDatabaseUserToUserModel(updatedUser);
```

Comment:
code style 💅: like my comment in the database, you could store some variables here and do this in a single call for cleanliness (and performance)

```ts
let nicknameUpdate: string | null = null;
let pictureUpdate: string | null = null;

if (nickname) {
    const trimmedNickname = nickname.trim();
    const nicknameError = await getNicknameError(filter, database, trimmedNickname);

    if (nicknameError) {
        res.status(400).send({
            type: "invalid_nickname",
            error: nicknameError
        });
        return;
    }

    nicknameUpdate = trimmedNickname;
}

if (picture) {
   /* ... */
}

const updatedUser = await database.user.setProfileInfo(user.id, nicknameUpdate, pictureUpdate);
const outputUser = convertDatabaseUserToUserModel(updatedUser);

// update metadata if anything changed
if (outputUser !== user) { /* ... */ }